src/main/kotlin/com/openlattice/search/renderers/CAREIssueAlertEmailRenderer.kt

```diff
@@ -0,0 +1,186 @@
+private val FIRST_NAME_FQN = FullQualifiedName("nc.PersonGivenName")
+private val MIDDLE_NAME_FQN = FullQualifiedName("nc.PersonMiddleName")
+private val LAST_NAME_FQN = FullQualifiedName("nc.PersonSurName")
+private val STATUS_FQN = FullQualifiedName("ol.status")
+private val DESCRIPTION_FQN = FullQualifiedName("ol.description")
+private val PRIORITY_FQN = FullQualifiedName("ol.priority")
+private val CATEGORY_FQN = FullQualifiedName("ol.category")
+private val TITLE_FQN = FullQualifiedName("ol.title")
+private val COMPLETED_DATE_TIME_FQN = FullQualifiedName("date.completeddatetime")
+private val ENTRY_UPDATED_FQN = FullQualifiedName("general.entryupdated")
+private val OPENLATTICE_ID_FQN = FullQualifiedName("openlattice.@id")
+
+
+/* METADATA TAGS */
+private const val PERSON_ENTITY_SET_ID_METADATA = "personEntitySetId"
+private const val STAFF_ENTITY_SET_ID_METADATA = "staffEntitySetId"
+private const val TIME_ZONE_METADATA = "timezone"
+private const val ASSIGNED_TO_ENTITY_SET_ID_METADATA = "assignedToEntitySetId"
+private const val REPORTED_ENTITY_SET_ID_METADATA = "reportedEntitySetId"
+
+/*
+*
+* Search metadata is expected to contain the following fields:
+* {
+*   "personEntitySetId": <UUID>,
+*   "staffEntitySetId": <UUID>,
+*   "assignedToEntitySetID": <UUID>,
+*   "reportedEntitySetID": <UUID>,
+*   "timezone": <TimeZones>,
+*   "alertName": <String>,
+* }
+*
+*/
+
+class CAREIssueAlertEmailRenderer {
+
+    companion object {
+
+        private fun isMatchByEntitySetId(neighbor :NeighborEntityDetails, entitySetId: UUID, associationEntitySetId: Optional<UUID>): Boolean {
+            val isMatchingEntitySetId = neighbor.neighborEntitySet.isPresent && neighbor.neighborEntitySet.get().id == entitySetId
+            return when (associationEntitySetId.isPresent) {
+                (neighbor.associationEntitySet.id == associationEntitySetId.get()) && isMatchingEntitySetId -> isMatchingEntitySetId
+                else -> isMatchingEntitySetId
+            }
+        }
+
+        private fun getCombinedNeighbors(
+                neighbors: List<NeighborEntityDetails>, entitySetId: UUID, associationEntitySetId: Optional<UUID>
+        ): Map<FullQualifiedName, Set<Any>> {
+            val entity = mutableMapOf<FullQualifiedName, MutableSet<Any>>()
+            neighbors.filter { isMatchByEntitySetId(it, entitySetId, associationEntitySetId) }.forEach {
+                it.neighborDetails.get().forEach { (fqn, values) ->
+                    entity.getOrPut(fqn) { mutableSetOf() }.addAll(values)
+                }
+            }
+
+            return entity
+        }
+
+        private fun getPersonDetails(
+                neighbors: List<NeighborEntityDetails>, personEntitySetId: UUID
+        ): Map<String, Any> {
+            val combinedEntity = getCombinedNeighbors(neighbors, personEntitySetId, Optional.empty())
+
+            val tags = mutableMapOf<String, String>()
```

Comment:
also pref sizing the map here though it may be simpler to return `Pair<String, String>` using `return "formattedName" to "$lastName, $firstName $middleName"`